### PR TITLE
[Form] [EventListener] fixed sending non array data on preSubmit to ResizeFormListener

### PR DESCRIPTION
**Pull request by BruceWouaigne**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #13804 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php [hunk 1]
```diff
@@ -145,7 +145,7 @@ public function onSubmit(FormEvent $event)
         }
 
         if (!is_array($data) && !($data instanceof \Traversable && $data instanceof \ArrayAccess)) {
-            throw new UnexpectedTypeException($data, 'array or (\Traversable and \ArrayAccess)');
+            $data = array();
```
Comment (ogizanagi) [inline src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php]: Shouldn't this be done in the `preSubmit` method instead ?

Comment (BruceWouaigne) [inline src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php]: Indeed

**Comment by Tobion** [comment]
Would seem more natural to me if `onSubmit` would also ignore invalid data.

**Comment by Tobion** [comment]
And it should be merged in 2.3

**Comment by xabbuh** [comment]
> Would seem more natural to me if `onSubmit` would also ignore invalid data.

`FormEvent::getData()` returns the normalised data in the `SUBMIT` event. Malformed data cannot be triggered by the request data, right? I don't think it makes sense to simply ignore it here.

**Comment by webmozart** [comment]
Thanks for working on this @BruceWouaigne! Looks good to me.

**Comment by BruceWouaigne** [comment]
Awesome! Thanks :)

**Comment by Tobion** [comment]
The other fixes of this kind have been merged into 2.3.

**Pull request closed without merging**